spec/annotation_spec.rb

```diff
@@ -0,0 +1,175 @@
+            it "[#{word}] truthy boolean" do
+              expect(Annotation.new("@probedock(active=#{quote}#{word}#{quote})").active).to be_truthy
+            end
+          end
+
+          %w(0 no false n f).each do |word|
+            it "[#{word}] falsey boolean" do
+              expect(Annotation.new("@probedock(active=#{quote}#{word}#{quote})").active).to be_falsey
+            end
+          end
+        end
+      end
+    end
+
+    %w[' "].each do |quote|
+      describe "annotation with spaces between quotes [#{quote}] and" do
+        describe 'key keyword' do
+          subject { Annotation.new("@probedock(key=#{quote}ab cde#{quote})") }
+          it 'should be possible' do
+            expect(subject.key).to eq('ab cde')
+          end
+        end
+
+        describe 'category keyword' do
+          subject { Annotation.new("@probedock(category=#{quote}ru by#{quote})") }
+          it 'should be possible' do
+            expect(subject.category).to eq('ru by')
+          end
+        end
+
+        describe('tag keyword') do
+          subject { Annotation.new("@probedock(tag=#{quote}ta g1#{quote})") }
+          it 'should be possible' do
+            expect(subject.tags).to eq(['ta g1'])
+          end
+        end
+
+        describe('ticket keyword') do
+          subject { Annotation.new("@probedock(ticket=#{quote}tick et1#{quote})") }
+          it 'should be possible' do
+            expect(subject.tickets).to eq(['tick et1'])
+          end
+        end
+      end
+    end
+
+    %w(ticket tag).each do |keyword|
+      describe "annotation with duplicated values for [#{keyword}]" do
+        subject { Annotation.new("@probedock(#{keyword}=val #{keyword}=val)") }
+        it 'should stored only once' do
+          expect(subject.send("#{keyword}s")).to eq([ 'val' ])
+        end
+      end
+    end
+
+    %w(key category).each do |keyword|
+      describe "latest encountered value for a #{keyword} in annotations" do
+        subject { Annotation.new("@probedock(#{keyword}=firstValue) @probedock(#{keyword}=lastValue)")}
+        it 'should be kept' do
+          expect(subject.send(keyword)).to eq('lastValue')
+        end
+      end
+    end
+
+    %w(ticket tag).each do |keyword|
+      describe "values for #{keyword} keywords in annotations" do
+        subject { Annotation.new("@probedock(#{keyword}=firstValue) @probedock(#{keyword}=lastValue)")}
+        it 'should be combined' do
+          expect(subject.send("#{keyword}s")).to eq( %w[firstValue lastValue])
+        end
+      end
+    end
+  end
+
+  describe '#merge' do
+    describe 'left is replaced by right for key, category and active and combined for tags and tickets' do
+      let(:left){ Annotation.new('@probedock(key=lkey category=lcat active=true tag=lta ticket=lti)') }
+      let(:right){ Annotation.new('@probedock(key=rkey category=rcat active=false tag=rta ticket=rti)') }
+      subject{ left.merge!(right) }
+      it {
```

Comment:
I don't much like your describe/it structure here. This is what the output looks like:

    #merge
      left is replaced by right for key, category and active and combined for tags and tickets
        should eq ["lti", "rti"]
      left is not replaced by right for key, category and active and not combined for tags and tickets when values are nil
        should eq ["lti"]

You're not reusing your `let` and `subject` blocks so I would remove them:

```
describe '#merge' do
  it "should replace the key, category and active flag and combine the tags and tickets"

    left = Annotation.new('@probedock(key=lkey category=lcat active=true tag=lta ticket=lti)')
    right = Annotation.new('@probedock(key=rkey category=rcat active=false tag=rta ticket=rti)')
    result = left.merge(right)

    expect(result.key).to eq('rkey')
    expect(result.category).to eq('rcat')
    expect(result.active).to be_falsey
    expect(result.tags).to eq(%w(lta rta))
    expect(result.tickets).to eq(%w(lti rti))
  end

  it "should not replace values when the merged annotation has none" do
    ...
  end
end
```